examples/research_projects/codeparrot/README.md

```diff
@@ -55,6 +55,16 @@ python scripts/preprocessing.py \
 ```
 During preprocessing the dataset is downloaded and stored locally as well as caches of the computations. Make sure you have more than 500GB free disk space to execute it.
 
+### Pretokenization
+The tokenization of the data might be slow during the training especially for small models. We provide code to pretokenize the data beforehand in `scripts/pretokenizing.py`, but we mention that this step is optional. The dataset is downloaded and stored locally but the tokenized data is directly pushed to the hub. The tokenized clean [train](https://huggingface.co/datasets/loubnabnl/tokenized-codeparrot-train) and [validation](https://huggingface.co/datasets/loubnabnl/tokenized-codeparrot-valid) datasets are available if you want to use them directly.
+
+To execute the pretokenization, for the clean train data for instance, run the following command:
+```bash
+python scripts/pretokenizing.py \
+--dataset_name lvwerra/codeparrot-clean-train \
+--tokenized_data_repo <your Hugging Face hub username>/tokenized-codeparrot-train
```

Comment:
If no username is supplied then it always pushes to your own account. No need to do this explicitly.

```suggestion
--tokenized_data_repo tokenized-codeparrot-train
```